Investigate why IT is failing

Travis CI is failing (at least last 2 builds) - need to investigate why. The error is below.

```
[INFO] --- maven-compiler-plugin:3.5.1:compile (default-compile) @ fluo-recipes-test ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 3 source files to /home/travis/build/apache/fluo-recipes/modules/test/target/classes
[INFO] -------------------------------------------------------------
[WARNING] COMPILATION WARNING : 
[INFO] -------------------------------------------------------------
[WARNING] /home/travis/build/apache/fluo-recipes/modules/test/src/main/java/org/apache/fluo/recipes/test/FluoITHelper.java:[30,39] org.apache.accumulo.core.client.Connector in org.apache.accumulo.core.client has been deprecated
[WARNING] /home/travis/build/apache/fluo-recipes/modules/test/src/main/java/org/apache/fluo/recipes/test/AccumuloExportITBase.java:[25,39] org.apache.accumulo.core.client.Connector in org.apache.accumulo.core.client has been deprecated
[WARNING] /home/travis/build/apache/fluo-recipes/modules/test/src/main/java/org/apache/fluo/recipes/test/AccumuloExportITBase.java:[202,13] org.apache.accumulo.core.client.Connector in org.apache.accumulo.core.client has been deprecated
[WARNING] /home/travis/build/apache/fluo-recipes/modules/test/src/main/java/org/apache/fluo/recipes/test/AccumuloExportITBase.java:[204,21] getConnector(java.lang.String,java.lang.String) in org.apache.accumulo.minicluster.MiniAccumuloCluster has been deprecated
[INFO] 4 warnings 
[INFO] -------------------------------------------------------------
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /home/travis/build/apache/fluo-recipes/modules/test/src/main/java/org/apache/fluo/recipes/test/FluoITHelper.java:[153,23] cannot find symbol
  symbol:   method since()
  location: @interface java.lang.Deprecated
[ERROR] /home/travis/build/apache/fluo-recipes/modules/test/src/main/java/org/apache/fluo/recipes/test/FluoITHelper.java:[153,45] cannot find symbol
  symbol:   method forRemoval()
  location: @interface java.lang.Deprecated
[ERROR] /home/travis/build/apache/fluo-recipes/modules/test/src/main/java/org/apache/fluo/recipes/test/FluoITHelper.java:[214,23] cannot find symbol
  symbol:   method since()
  location: @interface java.lang.Deprecated
[ERROR] /home/travis/build/apache/fluo-recipes/modules/test/src/main/java/org/apache/fluo/recipes/test/FluoITHelper.java:[214,45] cannot find symbol
  symbol:   method forRemoval()
  location: @interface java.lang.Deprecated
[INFO] 4 errors 
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Apache Fluo Recipes Parent 1.3.0-SNAPSHOT:
[INFO] 
[INFO] Apache Fluo Recipes Parent ......................... SUCCESS [01:25 min]
[INFO] Apache Fluo Recipes Core ........................... SUCCESS [05:00 min]
[INFO] Apache Fluo Recipes for Apache Accumulo ............ SUCCESS [ 10.865 s]
[INFO] Apache Fluo Recipes using Kryo ..................... SUCCESS [  8.077 s]
[INFO] Apache Fluo Recipes for Testing .................... FAILURE [  1.349 s]
[INFO] Apache Fluo Recipes for Apache Spark ............... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  06:57 min
[INFO] Finished at: 2020-10-05T21:05:24Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.5.1:compile (default-compile) on project fluo-recipes-test: Compilation failure: Compilation failure: 
[ERROR] /home/travis/build/apache/fluo-recipes/modules/test/src/main/java/org/apache/fluo/recipes/test/FluoITHelper.java:[153,23] cannot find symbol
[ERROR]   symbol:   method since()
[ERROR]   location: @interface java.lang.Deprecated
[ERROR] /home/travis/build/apache/fluo-recipes/modules/test/src/main/java/org/apache/fluo/recipes/test/FluoITHelper.java:[153,45] cannot find symbol
[ERROR]   symbol:   method forRemoval()
[ERROR]   location: @interface java.lang.Deprecated
[ERROR] /home/travis/build/apache/fluo-recipes/modules/test/src/main/java/org/apache/fluo/recipes/test/FluoITHelper.java:[214,23] cannot find symbol
[ERROR]   symbol:   method since()
[ERROR]   location: @interface java.lang.Deprecated
[ERROR] /home/travis/build/apache/fluo-recipes/modules/test/src/main/java/org/apache/fluo/recipes/test/FluoITHelper.java:[214,45] cannot find symbol
[ERROR]   symbol:   method forRemoval()
[ERROR]   location: @interface java.lang.Deprecated
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :fluo-recipes-test
The command "mvn clean verify javadoc:jar $ADDITIONAL_MAVEN_OPTS" exited with 1.
```

In cfc86ba4c1faf20da78f8a0fc149c737aca407c7 I think I introduced code that does not work w/ Java 8.  I suspect this is failing because java 8 is being used by Travis to build.  I just successfully ran mvn verify locally w/ Java 11.  The changes in #154 will use Java 11 to build, so that may fix this.   However, probably need to make make Java 11 the minimum in the pom.  I tried making the following change in the pom.xml locally

```diff
diff --git a/pom.xml b/pom.xml
index 0c604f1..638270b 100644
--- a/pom.xml
+++ b/pom.xml
@@ -56,8 +56,8 @@
     <findbugs.skip>true</findbugs.skip>
     <fluo.version>2.0.0-SNAPSHOT</fluo.version>
     <hadoop.version>3.1.0</hadoop.version>
-    <maven.compiler.source>1.8</maven.compiler.source>
-    <maven.compiler.target>1.8</maven.compiler.target>
+    <maven.compiler.source>11</maven.compiler.source>
+    <maven.compiler.target>11</maven.compiler.target>
     <releaseProfiles>fluo-recipes-release</releaseProfiles>
     <spark.version>2.4.6</spark.version>
     <zookeeper.version>3.4.8</zookeeper.version>
```

When I ran with these changes I saw the following error.

```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Apache Fluo Recipes Parent 1.3.0-SNAPSHOT:
[INFO] 
[INFO] Apache Fluo Recipes Parent ......................... SUCCESS [  3.791 s]
[INFO] Apache Fluo Recipes Core ........................... FAILURE [  2.628 s]
[INFO] Apache Fluo Recipes for Apache Accumulo ............ SKIPPED
[INFO] Apache Fluo Recipes using Kryo ..................... SKIPPED
[INFO] Apache Fluo Recipes for Testing .................... SKIPPED
[INFO] Apache Fluo Recipes for Apache Spark ............... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  6.586 s
[INFO] Finished at: 2020-10-08T17:17:14-04:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.gaul:modernizer-maven-plugin:1.5.0:modernizer (modernizer) on project fluo-recipes-core: Execution modernizer of goal org.gaul:modernizer-maven-plugin:1.5.0:modernizer failed.: IllegalArgumentException -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :fluo-recipes-core
```

May need to update the modernizer plugin version for Java 11, @ctubbsii  would you happen to know?

--- a/modules/core/src/main/java/org/apache/fluo/recipes/core/map/CollisionFreeMap.java
+++ b/modules/core/src/main/java/org/apache/fluo/recipes/core/map/CollisionFreeMap.java
@@ -22,7 +22,6 @@ import java.util.Map;
 
 import com.google.common.base.Preconditions;
 import com.google.common.base.Strings;
-import com.google.common.collect.ImmutableMap;
 import com.google.common.collect.Iterators;
 import com.google.common.hash.Hashing;
 import org.apache.fluo.api.client.SnapshotBase;
@@ -391,7 +390,7 @@ public class CollisionFreeMap<K, V> {
     opts.save(fluoConfig.getAppConfiguration());
 
     fluoConfig.addObserver(new org.apache.fluo.api.config.ObserverSpecification(
-        CollisionFreeMapObserver.class.getName(), ImmutableMap.of("mapId", opts.mapId)));
+        CollisionFreeMapObserver.class.getName(), Map.of("mapId", opts.mapId)));
   }
 
   /**

--- a/modules/core/src/test/java/org/apache/fluo/recipes/core/common/TestGrouping.java
+++ b/modules/core/src/test/java/org/apache/fluo/recipes/core/common/TestGrouping.java
@@ -19,7 +19,6 @@ import java.util.Set;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
-import com.google.common.collect.ImmutableSet;
 import org.apache.fluo.api.config.FluoConfiguration;
 import org.apache.fluo.api.config.SimpleConfiguration;
 import org.apache.fluo.api.data.Bytes;
@@ -69,8 +68,7 @@ public class TestGrouping {
     Assert.assertTrue(tableOptim.getSplits().contains(Bytes.of("m2:u:~")));
     Assert.assertTrue(tableOptim.getSplits().contains(Bytes.of("m2:d:~")));
 
-    Set<String> expectedGroups =
-        ImmutableSet.of("m1:u:", "m1:d:", "m2:u:", "m2:d:", "eq1:", "eq2:");
+    Set<String> expectedGroups = Set.of("m1:u:", "m1:d:", "m2:u:", "m2:d:", "eq1:", "eq2:");
 
     // ensure all splits group as expected
     for (Bytes split : tableOptim.getSplits()) {

--- a/modules/core/src/test/java/org/apache/fluo/recipes/core/map/it/BigUpdateIT.java
+++ b/modules/core/src/test/java/org/apache/fluo/recipes/core/map/it/BigUpdateIT.java
@@ -20,9 +20,9 @@ import java.util.HashMap;
 import java.util.Iterator;
 import java.util.Map;
 import java.util.Optional;
+import java.util.Set;
 import java.util.concurrent.atomic.AtomicInteger;
 
-import com.google.common.collect.ImmutableSet;
 import com.google.common.collect.MapDifference;
 import com.google.common.collect.Maps;
 import org.apache.commons.io.FileUtils;
@@ -99,7 +99,7 @@ public class BigUpdateIT {
 
       // get last values set to verify same as passed in old value
       Map<String, Long> actualOld = Maps.transformValues(
-          ttx.get().rowsString(expectedOld.keySet()).columns(ImmutableSet.of(DSCOL)).toStringMap(),
+          ttx.get().rowsString(expectedOld.keySet()).columns(Set.of(DSCOL)).toStringMap(),
           m -> m.get(DSCOL).toLong());
 
       MapDifference<String, Long> diff = Maps.difference(expectedOld, actualOld);

--- a/modules/core/src/test/java/org/apache/fluo/recipes/core/map/it/CollisionFreeMapIT.java
+++ b/modules/core/src/test/java/org/apache/fluo/recipes/core/map/it/CollisionFreeMapIT.java
@@ -20,7 +20,6 @@ import java.util.HashMap;
 import java.util.Map;
 import java.util.Random;
 
-import com.google.common.collect.ImmutableMap;
 import org.apache.commons.io.FileUtils;
 import org.apache.fluo.api.client.FluoClient;
 import org.apache.fluo.api.client.FluoFactory;
@@ -128,17 +127,17 @@ public class CollisionFreeMapIT {
   public void testGet() {
     try (FluoClient fc = FluoFactory.newClient(miniFluo.getClientConfiguration())) {
       try (Transaction tx = fc.newTransaction()) {
-        wcMap.update(tx, ImmutableMap.of("cat", 2L, "dog", 5L));
+        wcMap.update(tx, Map.of("cat", 2L, "dog", 5L));
         tx.commit();
       }
 
       try (Transaction tx = fc.newTransaction()) {
-        wcMap.update(tx, ImmutableMap.of("cat", 1L, "dog", 1L));
+        wcMap.update(tx, Map.of("cat", 1L, "dog", 1L));
         tx.commit();
       }
 
       try (Transaction tx = fc.newTransaction()) {
-        wcMap.update(tx, ImmutableMap.of("cat", 1L, "dog", 1L, "fish", 2L));
+        wcMap.update(tx, Map.of("cat", 1L, "dog", 1L, "fish", 2L));
         tx.commit();
       }
 
@@ -167,7 +166,7 @@ public class CollisionFreeMapIT {
       Assert.assertEquals(expectedCounts, getComputedWordCounts(fc));
 
       try (Transaction tx = fc.newTransaction()) {
-        wcMap.update(tx, ImmutableMap.of("cat", 1L, "dog", -7L));
+        wcMap.update(tx, Map.of("cat", 1L, "dog", -7L));
         tx.commit();
       }
 

--- a/modules/core/src/test/java/org/apache/fluo/recipes/core/types/TypeLayerTest.java
+++ b/modules/core/src/test/java/org/apache/fluo/recipes/core/types/TypeLayerTest.java
@@ -18,8 +18,8 @@ package org.apache.fluo.recipes.core.types;
 import java.nio.ByteBuffer;
 import java.util.Arrays;
 import java.util.Map;
+import java.util.Set;
 
-import com.google.common.collect.ImmutableSet;
 import org.apache.fluo.api.data.Bytes;
 import org.apache.fluo.api.data.Column;
 import org.apache.fluo.recipes.core.types.TypedSnapshotBase.Value;
@@ -38,8 +38,8 @@ public class TypeLayerTest {
 
     TypedTransactionBase ttx = tl.wrap(tt);
 
-    Map<Column, Value> results = ttx.get().row("r2")
-        .columns(ImmutableSet.of(new Column("cf2", "6"), new Column("cf2", "7")));
+    Map<Column, Value> results =
+        ttx.get().row("r2").columns(Set.of(new Column("cf2", "6"), new Column("cf2", "7")));
 
     Assert.assertNull(results.get(new Column("cf2", "6")).toInteger());
     Assert.assertEquals(0, results.get(new Column("cf2", "6")).toInteger(0));
@@ -48,8 +48,8 @@ public class TypeLayerTest {
 
     Assert.assertEquals(1, results.size());
 
-    results = ttx.get().row("r2").columns(
-        ImmutableSet.of(new Column("cf2", "6"), new Column("cf2", "7"), new Column("cf2", "8")));
+    results = ttx.get().row("r2")
+        .columns(Set.of(new Column("cf2", "6"), new Column("cf2", "7"), new Column("cf2", "8")));
 
     Assert.assertNull(results.get(new Column("cf2", "6")).toInteger());
     Assert.assertEquals(0, results.get(new Column("cf2", "6")).toInteger(0));
@@ -450,20 +450,20 @@ public class TypeLayerTest {
     Assert.assertNull(ttx.get(Bytes.of("r2"), new Column("cf2", "9")));
 
     Map<Column, Bytes> map =
-        ttx.get(Bytes.of("r2"), ImmutableSet.of(new Column("cf2", "7"), new Column("cf2", "8")));
+        ttx.get(Bytes.of("r2"), Set.of(new Column("cf2", "7"), new Column("cf2", "8")));
     Assert.assertEquals(2, map.size());
     Assert.assertEquals("12", map.get(new Column("cf2", "7")).toString());
     Assert.assertEquals("13", map.get(new Column("cf2", "8")).toString());
 
-    map = ttx.get(Bytes.of("r6"), ImmutableSet.of(new Column("cf2", "7"), new Column("cf2", "8")));
+    map = ttx.get(Bytes.of("r6"), Set.of(new Column("cf2", "7"), new Column("cf2", "8")));
     Assert.assertEquals(0, map.size());
 
     ttx.set(Bytes.of("r6"), new Column("cf2", "7"), Bytes.of("3"));
     Assert.assertEquals(MockTransactionBase.toRCVM("r6,cf2:7,3"), tt.setData);
     tt.setData.clear();
 
-    Map<Bytes, Map<Column, Bytes>> map2 = ttx.get(ImmutableSet.of(Bytes.of("r1"), Bytes.of("r2")),
-        ImmutableSet.of(new Column("cf1", "cq1"), new Column("cf2", "8")));
+    Map<Bytes, Map<Column, Bytes>> map2 = ttx.get(Set.of(Bytes.of("r1"), Bytes.of("r2")),
+        Set.of(new Column("cf1", "cq1"), new Column("cf2", "8")));
     Assert.assertEquals(MockTransactionBase.toRCVM("r1,cf1:cq1,v1", "r2,cf2:8,13"), map2);
 
     ttx.delete(Bytes.of("r6"), new Column("cf2", "7"));

--- a/modules/spark/src/main/java/org/apache/fluo/recipes/spark/FluoSparkHelper.java
+++ b/modules/spark/src/main/java/org/apache/fluo/recipes/spark/FluoSparkHelper.java
@@ -278,7 +278,7 @@ public class FluoSparkHelper {
      * FluoConfiguration supplied to
      * {@link FluoSparkHelper#FluoSparkHelper(FluoConfiguration, Configuration, Path)}
      *
-     * @param conn Use this connector to bulk import files into Accumulo.
+     * @param client Use this client to bulk import files into Accumulo.
      * @return this
      *
      * @since 1.3.0

--- a/pom.xml
+++ b/pom.xml
@@ -56,9 +56,9 @@
     <findbugs.skip>true</findbugs.skip>
     <fluo.version>2.0.0-SNAPSHOT</fluo.version>
     <hadoop.version>3.1.0</hadoop.version>
-    <maven.compiler.release>8</maven.compiler.release>
-    <maven.compiler.source>1.8</maven.compiler.source>
-    <maven.compiler.target>1.8</maven.compiler.target>
+    <maven.compiler.release>11</maven.compiler.release>
+    <maven.compiler.source>11</maven.compiler.source>
+    <maven.compiler.target>11</maven.compiler.target>
     <releaseProfiles>fluo-recipes-release</releaseProfiles>
     <spark.version>2.4.6</spark.version>
     <zookeeper.version>3.4.8</zookeeper.version>
@@ -226,6 +226,19 @@
     <pluginManagement>
       <plugins>
         <plugin>
+          <groupId>org.gaul</groupId>
+          <artifactId>modernizer-maven-plugin</artifactId>
+          <version>1.8.0</version>
+          <configuration>
+            <javaVersion>${maven.compiler.target}</javaVersion>
+          </configuration>
+        </plugin>
+        <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-enforcer-plugin</artifactId>
+          <version>3.0.0-M3</version>
+        </plugin>
+        <plugin>
           <groupId>org.apache.maven.plugins</groupId>
           <artifactId>maven-compiler-plugin</artifactId>
           <version>3.8.1</version>
@@ -267,6 +280,11 @@
               <fluo.it.instance.clear>false</fluo.it.instance.clear>
             </systemPropertyVariables>
           </configuration>
+        </plugin>
+        <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-dependency-plugin</artifactId>
+          <version>3.1.2</version>
         </plugin>
       </plugins>
     </pluginManagement>

